pkg/controller/workspacerouting/solvers/oauth_proxy_container.go

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package solvers
+
+import (
+	"fmt"
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/common"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+)
+
+const proxyServiceAcctAnnotationKeyFmt string = "serviceaccounts.openshift.io/oauth-redirectreference.%s-%s"
+const proxyServiceAcctAnnotationValueFmt string = `{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"%s"}}`
+
+var openShiftProxySARFmt = `{"namespace": "%s", "resource": "pods", "name": "%s", "verb": "exec"}`
+
+func getProxyPodAdditions(proxyEndpoints map[string]proxyEndpoint, meta WorkspaceMetadata) *v1alpha1.PodAdditions {
+	var proxyContainers []corev1.Container
+	for _, proxyEndpoint := range proxyEndpoints {
+		proxyContainers = append(proxyContainers, getProxyContainerForEndpoint(proxyEndpoint, meta))
+	}
+	proxyVolumes := getProxyVolumes(proxyContainers)
+	serviceAcctAnnotations := getProxyServiceAcctAnnotations(proxyEndpoints, meta)
+
+	return &v1alpha1.PodAdditions{
+		Containers:                proxyContainers,
+		Volumes:                   proxyVolumes,
+		ServiceAccountAnnotations: serviceAcctAnnotations,
+	}
+}
+
+func getProxyContainerForEndpoint(proxyEndpoint proxyEndpoint, meta WorkspaceMetadata) corev1.Container {
+	proxyContainerName := fmt.Sprintf("%s-oauth-proxy-%s", meta.WorkspaceId, strconv.FormatInt(proxyEndpoint.upstreamEndpoint.Port, 10))
+
+	return corev1.Container{
+		Name: proxyContainerName,
+		Ports: []corev1.ContainerPort{
+			{
+				//Name:          endpoint.upstreamEndpoint.Name,
+				ContainerPort: int32(proxyEndpoint.publicEndpoint.Port),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		ImagePullPolicy: corev1.PullPolicy(config.ControllerCfg.GetSidecarPullPolicy()),
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "proxy-tls",
```

Comment:
to be consist in proposal in https://github.com/che-incubator/che-workspace-operator/pull/36/files#r403919074
```suggestion
				Name:      "proxy-tls-" + meta.WorkspaceId,
```


P.S. It surprised me a bit that volumeMounts is used as secret name. It's OK for time being but it would be better to design more straightforward solution. Something like https://github.com/che-incubator/che-workspace-operator/commit/117c52150aade53fb90e0d0e329908f22333781f

Reply:
Fixed :)